invenio_app_rdm/utils/checks.py

```diff
@@ -65,12 +65,18 @@ def resolve_checks(record_uuid, request, community=None):
 
     # Find check runs for the given check configs
     check_config_ids = [check_config.id for check_config in check_configs]
-    checks = CheckRun.query.filter(
-        CheckRun.config_id.in_(check_config_ids),
-        CheckRun.record_id == record_uuid,
-    ).all()
-    # For a given record, there is one check run corresponding to one check config
-    # Order the check runs by the same order as the check configs for deterministic ordering
-    checks = sorted(checks, key=lambda check: check_config_ids.index(check.config_id))
-
-    return checks
+
+    check_runs = (
+        CheckRun.query.filter(
+            CheckRun.config_id.in_(check_config_ids), CheckRun.record_id == record_uuid
+        )
+        .order_by(CheckRun.is_draft.desc(), CheckRun.revision_id.desc())
```

Comment:
I have the same feeling here, as we had in the Checks component. We should be very explicit about which check runs we're fetching, i.e. for the record or draft. Either that, or we fetch them all and sort/split them out here in code (if e.g. we plan to be able to display them both). Doing "tricks" with sorting based on `is_draft` and `revision_id` can be confusing and faulty.